packages/core/src/internal/base/button.base.ts

```diff
@@ -83,6 +85,20 @@ export class CdsBaseButton extends LitElement {
   protected updated(props: Map<string, any>) {
     super.updated(props);
     this.updateButtonAttributes();
+
+    if (props.has('disabled')) {
+      const isDisabled = this.disabled;
+      if (isDisabled !== (this.ariaDisabled === 'true')) {
+        this.ariaDisabled = isDisabled ? 'true' : 'false';
+      }
+    }
+
+    if (props.has('ariaDisabled')) {
+      const isAriaDisabled = this.ariaDisabled === 'true';
+      if (this.disabled !== isAriaDisabled) {
+        this.disabled = isAriaDisabled ? true : false;
+      }
+    }
```

Comment:
Minor cleanup nit, could abstract this part into a method something like `syncDisabledProperties(props)`

Reply:
This got merged already. But I can look into abstracting it in a separate cleanup. That is a good idea.